NOTES.WIN

```diff
@@ -83,6 +83,18 @@
  is, of course, to choose a different set of directories by using
  --prefix and --openssldir when configuring.
 
+
+ Special notes for Universal Windows Platform builds, a.k.a. VC-*-UWP
+ --------------------------------------------------------------------
+
+ - UWP targets only support building the static and dynamic libraries.
+
+ - The "no-asm" and "no-uplink" must be given in the "Configure" script.
```

Comment:
Why must no-asm be given?

Reply:
https://github.com/openssl/openssl/pull/7729#issuecomment-448982681

I have asked the question before and have an answer about that. Hope it can help you.

Reply:
IMO this is a significant issue and drastically reduces the usefulness of this PR. Not only is there a potentially huge performance hit, but also much of the C code is not constant time whereas the assembler is. I don't really consider "no-asm" to be an option we should be encouraging for production.

Reply:
Microsoft's OpenSSL branch for UWP uses "no-asm" option. But I will try to compile it without "no-asm" option ASAP. (I will remove it if OK.)

Reply:
Without `no-asm`, there are a gazillion messages complaining about the lack of `__imp_RtlVirtualUnwind` when building for VC-WIN64A-UWP...

Reply:
So, it seems that issue should be addressed before this PR should be accepted.

Reply:
@levitte 
Which version of Windows 10 SDK you are used? Because we can use RtlVirtualUnwind in Windows 10 RS5 or later. (https://docs.microsoft.com/en-us/uwp/win32-and-com/win32-extension-apis#apis-from-api-ms-win-core-rtlsupport-l1-2-0dll)

I also want to show you a talk between the maintainer from Microsoft/OpenSSL and me.

> Also I think we should keep the compatibility for earlier version of Windows, even compiled with the latest SDK. Because some developers like me need to support several versions of Windows 10.

> Building with a new SDK still allows the binary to run on previous versions of Windows, so moving forward with the progress of the SDK doesn't prevent supporting older versions.

> But after reading the source code, I realized that some functions will be linked to the new version of dynamic libraries if I am using the new version of Windows SDK because you used the conditional compilation. (It will not overwrite functions not supported by older versions if the new version of Windows SDK is used.)

> Yes, while when using a new SDK we might be linking around newly exposed functions (specifically the ones that we override), they are already part of the previous versions of the OS. The apis weren't exposed in the Universal SDK and would fail the WACK certification, but as the SDK and WACK allow for more functions to be used they are also allowed on previous versions of the OS.

For more informaton, please read https://github.com/microsoft/openssl/issues/59

Thank you.

Kenji Mouri

Reply:
@mattcaswell I have tried with Windows 10 SDK Build 18362, and find OK without "no-asm" option. But you will meet the problem which @levitte founded in Windows 10 SDK Build 17134 or earlier. So should I say "You also should add "no-asm" option when you use the Windows 10 SDK Build 17134 or earlier" ?

Reply:
I'd say "no" on the basis that we shouldn't be encouraging use of "no-asm". I'd suggest you drop all mention and use of no-asm from this PR, and then (once this PR is merged) raise a separate issue about it not working on earlier Windows 10 SDK Builds. We can then figure out whether there is a better answer than recommending no-asm. Or possibly we should say those earlier builds are not supported.

Reply:
I have followed your suggestion. @mattcaswell